### PR TITLE
Add steps for setting up new translations on Weblate

### PR DESCRIPTION
**Pull request by amyblais**
Based on https://gist.github.com/streamer45/0021066aeaee50abbee508ddfa50903e

### CODE REVIEW

File: contributors/join-us/localization.md [hunk 1]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+## Setting up new translations on Weblate
+
+### Step 0 - Permissions
+
+Register an account and get admin permissions to the [Weblate server](https://translate.mattermost.com/).
+
+### Step 1 -  Create a new Project (optional)
+
+This step makes since if the project has its own repository (e.g. Calls) or if it's helpful to have a separate translation entry point. Other option is to simply skip this step and to add a new component to an existing project.
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
This step makes sense when the project has its own repository (e.g. Calls) or if it's helpful to have a separate translation entry point. Another option is to simply skip this step, and add a new component to an existing project.
```

File: contributors/join-us/localization.md [hunk 2]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+**Note**
+
+Make sure to enable reviews for the project. Go to project settings, workflow tab and select enable reviews:
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]: This note may read better if both the note prefix and the note body are together on the same line.

Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
Make sure to enable reviews for the project. In Weblate, go to **Project Settings**, select the **Workflow** tab, and select **Enable Reviews**:
```

File: contributors/join-us/localization.md [hunk 3]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+![image](https://user-images.githubusercontent.com/1832946/227036224-cb42bbd5-f101-4d42-b18d-fe5d8f3f44c4.png)
+
+### Step 2 Add component(s)
+
+Add a new component under the project
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
Add a new component under the project.
```

File: contributors/join-us/localization.md [hunk 4]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
 2. Select a component on the resulting page, then select **Tools > Start new translation**. 
 3. Select the languages you want to add, then **Start new translation**.
 4. Repeat step 2 for every component within the [https://translate.mattermost.com/projects/i18n-wip](https://translate.mattermost.com/projects/i18n-wip) project.
+
+## Setting up new translations on Weblate
+
+### Step 0 - Permissions
+
+Register an account and get admin permissions to the [Weblate server](https://translate.mattermost.com/).
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
Register an account and request admin permissions to the [Weblate server](https://translate.mattermost.com/) from Carrie Warner or Tom De Moor.
```

File: contributors/join-us/localization.md [hunk 5]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+### Step 1 -  Create a new Project (optional)
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
### Step 1 - Create a new Project (optional)
```

File: contributors/join-us/localization.md [hunk 6]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+This step makes since if the project has its own repository (e.g. Calls) or if it's helpful to have a separate translation entry point. Other option is to simply skip this step and to add a new component to an existing project.
+
+**Note**
+
+Make sure to enable reviews for the project. Go to project settings, workflow tab and select enable reviews:
+
+![image](https://user-images.githubusercontent.com/1832946/227036224-cb42bbd5-f101-4d42-b18d-fe5d8f3f44c4.png)
+
+### Step 2 Add component(s)
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
### Step 2 - Add component(s)
```

File: contributors/join-us/localization.md [hunk 7]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+Add a new component under the project
+
+![image](https://user-images.githubusercontent.com/1832946/227037904-2165a769-0d6d-4b4e-8b6e-2260c2061d60.png)
+
+Base configuration should be like the following:
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
Base configuration should look as follows:
```

File: contributors/join-us/localization.md [hunk 8]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+- *Version control system* should be set to *Github pull request*
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
- *Version control system* should be set to *Github pull request*.
```

File: contributors/join-us/localization.md [hunk 9]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+- *Source code repository* should be set to the GitHub repository where the translation files are stored
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
- *Source code repository* should be set to the GitHub repository where the translation files are stored.
```

File: contributors/join-us/localization.md [hunk 10]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+- *Repository branch* should be set to the primary branch where translations should get merged into (usually either `master` or `main`).
+
+Upon clicking on continue, Weblate will attempt to scan the repository to find any existing translation files paths. You can now select the one you'd like to cover with this component.
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
When you select **Continue**, Weblate will attempt to scan the repository to find any existing translation files paths. You can now select the one you'd like to cover with this component.
```

File: contributors/join-us/localization.md [hunk 11]
```diff
@@ -207,3 +207,49 @@ To add a new WIP language:
...
+
+Uncheck *Edit base file* as it will prevent contributors from modifying the default language file and prevent potential conflicts.
+
+![image](https://user-images.githubusercontent.com/1832946/227040250-2dec9c1e-da44-41b4-ae12-2ebe3f85681d.png)
+
+Set *Translation flags* to `ignore-inconsistent, ignore-same, icu-message-format`
```
Comment (cwarnermm) [inline contributors/join-us/localization.md]:
```suggestion
Set *Translation flags* to `ignore-inconsistent, ignore-same, icu-message-format`.
```

**Review by cwarnermm — DISMISSED** [review]
Thank you both for this great Handbook PR. I've provided editorial feedback inline to consider. Approving for expediency.

File: contributors/join-us/localization.md [hunk 12]
```diff
@@ -207,3 +207,47 @@ To add a new WIP language:
...
+
+![image](https://user-images.githubusercontent.com/1832946/227045392-ca3640bd-d0c1-48d2-86ee-657b40e068b2.png)
+
+### Step 3 - Repository hooks
+
+Repository hooks (usually GitHub hooks) are needed to keep the translation files on Weblate in sync with the ones commited to the project repository. 
+
+![image](https://user-images.githubusercontent.com/1832946/227048773-db198f4b-4f73-4e20-a89c-a1b8492bf0df.png)
+
+You can create a ticket for the Release team if you need Admin permissions on the repository.
```
Comment (streamer45) [inline contributors/join-us/localization.md]: Just to note I am not sure they are allowed to give you admin permissions so it's more likely they will enable the hooks for you instead. Result is the same in this context as it's a one time setup anyway. Maybe @spirosoik can confirm what's the best approach here.

**Review by streamer45 — DISMISSED** [review]
Thank you both :100:

File: contributors/join-us/localization.md (hunk 12 above)
Comment (spirosoik) [reply contributors/join-us/localization.md]: Agree with Claudio we are trying to keep “least privilege" basis to ensure that roles have the minimum permissions required and are periodically re-evaluated

**Review by cwarnermm — APPROVED** [review]
Thank you, @amyblais and @streamer45!

**Pull request merged**